test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -10884,6 +10886,35 @@ def log_match(self, msg=None, id=None, n=50, tail=True, allmatch=False,
                                max_attempts, interval, starttime, endtime,
                                level=level, existence=existence)
 
+    def run_scheduling_cycle(self):
+        """
+        Convenience method to start and finish a sched cycle
+        """
+        sched = self.attributes['id']
+        old_val = self.server.status(SCHED, 'scheduling', id=sched)[
+            0]['scheduling']
+
+        # Make sure that we aren't in a sched cycle already
+        self.server.manager(MGR_CMD_SET, SCHED, {
+                            'scheduling': 'False'}, id=sched)
```

Comment:
would this stop the already running cycle?

Reply:
no, it will just make sure that IF there's an existing cycle running, we'll wait until it's done